### PR TITLE
Add mapper and disable auto extraction for ThinkstCanary

### PR DESCRIPTION
**Pull request by amshamah419**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-3602

#### Incident Fields

- New: **Thinkst Canary Flock Id**
- New: **Thinkst Canary Node Id**
- New: **Thinkst Canary Flock Name**

#### Incident Types

- New: **Thinkst Canary Incident**

##### Thinkst Canary

- Removed the ThinkstCanary URL from the human readable outputs of the ***canarytools-list-tokens*** and ***canarytools-get-token*** commands.
- Updated the Docker image to: *demisto/python3:3.10.13.73190*.

##### New: Thinkst Canary - Incoming Mapper

- New: Map fields from incoming Thinkst Canary events. (Available from Cortex XSOAR 6.9.0).

##### New: Enrich ThinkstCanary Events

- New: Enrich events that are received as part of a ThinkstCanary incident. (Available from Cortex XSOAR 6.9.0).

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by amshamah419** [comment]
The tests are failing due to some abnormal behavior by the fetch incidents test mechanism. Fetching does work correctly so I'm not sure this is a blocker to merge.

File: Packs/ThinkstCanary/Integrations/ThinkstCanary/ThinkstCanary.py [hunk 1]
```diff
@@ -195,9 +195,12 @@ def list_tokens_command():
     context = createContext(new_tokens, removeNull=True)
 
     contents = res_json
-    human_readable = tableToMarkdown('Canary Tools Tokens', new_tokens, headers=headers)
+    for token in new_tokens:
+        if 'TokenURL' in token:
+            del token['TokenURL']
```
Comment (dorschw) [inline Packs/ThinkstCanary/Integrations/ThinkstCanary/ThinkstCanary.py:200]:
```suggestion
        token.pop('TokenURL', None)
```

File: Packs/ThinkstCanary/Integrations/ThinkstCanary/ThinkstCanary.py [hunk 2]
```diff
@@ -226,9 +229,12 @@ def get_token_command():
         token_file = fileResult(name, content)
         demisto.results(token_file)
     else:
-        human_readable = tableToMarkdown('Canary Tools Tokens', res.get('token'))
+        token_data = res.get('token')
+        if 'url' in token_data:
+            del token_data['url']
```
Comment (dorschw) [inline Packs/ThinkstCanary/Integrations/ThinkstCanary/ThinkstCanary.py:234]:
```suggestion
        token_data.pop('url', None)
```
Thanks for inspiring [this](https://github.com/astral-sh/ruff/issues/7537)

File: Packs/ThinkstCanary/Integrations/ThinkstCanary/ThinkstCanary.py [hunk 3]
```diff
@@ -226,9 +229,12 @@ def get_token_command():
         token_file = fileResult(name, content)
         demisto.results(token_file)
     else:
-        human_readable = tableToMarkdown('Canary Tools Tokens', res.get('token'))
+        token_data = res.get('token')
```
Comment (dorschw) [inline Packs/ThinkstCanary/Integrations/ThinkstCanary/ThinkstCanary.py:232]: if we need the `.get` over direct accessors (i.e. key may not be presend), add a dictionary as default.
if we do not, replace it with `[]` access please.

**Review by dorschw — COMMENTED** [review]
nice

File: Packs/ThinkstCanary/Integrations/ThinkstCanary/ThinkstCanary.py (hunk 2 above)
Comment (amshamah419) [reply Packs/ThinkstCanary/Integrations/ThinkstCanary/ThinkstCanary.py:234]: oylo

**Review by dorschw — APPROVED** [review]

**Pull request merged**